### PR TITLE
feat: allow objects to be passed from STDIN.

### PR DESCRIPTION
**Pull request by jhoover4**
### Description
PR to address feature change requested in https://github.com/IBM/argocd-vault-plugin/issues/116.

Open to suggestions/review!

**Fixes:** 
https://github.com/IBM/argocd-vault-plugin/issues/116

### Checklist
Please make sure that your PR fulfills the following requirements:
- [x] Reviewed the guidelines for contributing to this repository
- [x] The commit message follows the [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] Tests for the changes have been updated
- [x] Docs have been added / updated

### Type of Change

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

### CODE REVIEW

File: cmd/generate.go
Comment (jhoover4) [inline cmd/generate.go:58]: I update this because there could be a potential naming conflict with the package name above

File: cmd/util.go
Comment (jhoover4) [inline cmd/util.go:58]: Didn't think the path was important here, but open to finding a way to add it back if they did pass a file

Comment (jkayani) [reply cmd/util.go:58]: I think it'd be good to have since the plugin can be operate on all the files in a directory, so if one is malformed it'd be obvious which one

**Review by jkayani — COMMENTED** [review]
Looks great, especially the tests! If you can find a way to keep the `YAML file at /blah/blah could not be read` message that'd be great. If not, I don't think it's a deal breaker

File: cmd/util.go
Comment (jhoover4) [reply cmd/util.go:58]: Cool I think I can come up with something!

**Review by werne2j — APPROVED** [review]
Looks good!

**Comment by werne2j** [comment]
Will wait for @jkayani to approve before merging

File: cmd/generate_test.go
Comment (jkayani) [inline cmd/generate_test.go:167]: This is a helpful enough message so that someone using it from STDIN has an idea of where it went wrong, good enough for me :)

**Review by jkayani — APPROVED** [review]
Thanks for the contribution @jhoover4 !

**Comment by jkayani** [comment]
@werne2j I will let you merge and decide when to release (1.1 I guess)

**Comment by jhoover4** [comment]
Awesome! Let me know when we release it I'd like to update the script I made to use stdin :)

**Pull request merged**
